src/graph/graphQLClient.ts

```diff
@@ -17,13 +17,23 @@ import config from '../config';
  * gives a graphQLClient for pocket-graph url
  *
  * This client initializes a `graphql-request` client
- * @param access_token accessToken of the user
- * @param consumer_key consumerKey assocuated with the user
+ * @param headers any headers received by proxy is just pass through to web graphQL proxy.
+ * @param accessToken accessToken of the user
+ * @param consumerKey consumerKey associated with the user
  */
-export function getClient(accessToken: string, consumerKey: string) {
+export function getClient(
+  accessToken: string,
+  consumerKey: string,
+  headers: any
+) {
   return new GraphQLClient(
     `${config.graphQLProxy}?consumer_key=${consumerKey}&access_token=${accessToken}`,
     {
+      //todo : investigate: ideally, we should pass any headers.
+      //however this fails if we dont explicitly pass cookie header
+      headers: {
+        cookie: headers?.cookie,
```

Comment:
@bassrock do we need all headers instead of just passing `cookie`?

Reply:
@sri-kirsh yes we should pass through all headers